legion/legion/serving/pyserve.py

```diff
@@ -281,6 +282,26 @@ def init_model(application):
                     'or model does not contain required props')
 
 
+class CustomJSONEncoder(json.JSONEncoder):
+    def default(self, o):  # pylint: disable=E0202
+        """
+        Redefined method, allows json to encode values of types np.integer, np.float, pd.Series
+        :param o: np.floating, np.integer
+        :return: float, int, List[int], List[float]
+        """
+        if isinstance(o, pd.Series):
+            values_list = o.tolist()
+            # return single value, because int(pd.Series) returns `int` if pd.Series contains single element
```

Comment:
I'm not quite sure about that. @mcdoker18 what do you think?

Reply:
I have added this because we were actually casting `pd.Series` with single value to int in `test_model_invoke_summation_with_df_with_prepare`.
It was casted here https://github.com/legion-platform/legion/pull/796/files#diff-893cca846985cb4a258dbd5623ae1532L39

Reply:
This line is actually not needed, but left it just in case https://github.com/legion-platform/legion/pull/796/files#diff-ee1a900f79e439fb7e5d715908976443R297

Reply:
Sorry for delay. I agree with comment above return. If a pandas series with one element can cast to int, we should keep it.